core/packages/contracts/src/Assets.sol

```diff
@@ -74,7 +76,7 @@ library Assets {
 
         _transferToAgent(assetHubAgent, token, sender, amount);
 
-        payload = SubstrateTypes.SendToken(assetHubAgent, token, destinationChain, destinationAddress, amount);
+        payload = SubstrateTypes.SendToken(gateway, token, destinationChain, destinationAddress, amount);
```

Comment:
nice catch! Though it's not necessary to pass `gateway` as a parameter to `Assets.sendToken`. Since `Assets.sol` is a library used by `Gateway`, you can just get the address of the gateway via `address(this)`.

```suggestion
        payload = SubstrateTypes.SendToken(address(this), token, destinationChain, destinationAddress, amount);
```

Reply:
addressed in https://github.com/Snowfork/snowbridge/pull/904/commits/8289362594510c9672b87bd44bfbf09dcbe3f596